File: Mine/05Monopoly/Monopoly-Q2特别版/src/map/Map.java
package map;

import java.util.Vector;

import block.Block;

public abstract class Map {
    protected MapImp mapImp;
    protected String name;

	public Map(MapImp aImp) {
		mapImp = aImp;
	}

	/**
	* show Map
	*/
    public void show() {
        mapImp.show();
    }

	public void addBlock(Block block) {
		mapImp.addBlock(block);
	}

	public Block getBlock(int index){
		return mapImp.getBlock(index);
	}

	public int getRow() {
		return mapImp.getRow();
	}

	public int getCol() {
		return mapImp.getCol();
	}

	public void setBlockVector(Vector<Block> blockVector) {
		mapImp.setBlockVector(blockVector);
	}
	
	public Vector<Block> getBlockVector() {
		return mapImp.getblockVector();
	}	
}

File: Teacher/04Richer/作业03/step2_src_完成123/map.cpp
///======================================================================
///  Project:   Richer02
/// FileName:	map.cpp
///     Desc:   Richer 02
///   Author:	Chen Wei
///======================================================================
#include <iostream>
using namespace std;
#include "global.h"
#include "block.h"

#include "map.h"
#include "mapimp.h"

Map::Map(MapImp* aImp)
    : imp(aImp)
{
}

Map::~Map()
{
    delete imp;
}
void Map::addBlock(Block* block)
{
    imp->addBlock(block);
}
Block* Map::getBlock(int index)
{
    return imp->getBlock(index);
}
void Map::show() const
{
    imp->show();
    /**
    for(int r = 0; r < BlockID::BLOCK_ROWS; ++r) {
        cout << endl;
        for(int c = 0; c < BlockID::BLOCK_COLS; ++c) {
            Block* block = blocks[r][c];
            if(block) {
                cout << block->name() << " ";
            } else {
                cout << "  ";
            }
        }
    }
    cout << endl;
    **/
}
